src/helpers/prf_compute/multiplicative_inverse.rs

```diff
@@ -0,0 +1,84 @@
+use num_bigint::{BigInt, BigUint, ToBigInt};
+use num_traits::{identities::Zero, One, Signed};
+
+/// Implementation of Extended Euclidean Algorithm to compute multiplicative inverse of a number
+/// i.e given num and modulus it computes t such that (num * t) % modulus = 1
+///
+/// How it computes :
+/// Finds Bezout Coefficient a and b for coprimes num and modulus i.e.
+/// ``bezout_coefficient_a`` * ``num`` + ``bezout_coefficient_b`` * ``modulus`` = 1
+/// This deduces to ( ``bezout_coefficient_a`` * ``num`` ) % ``modulus`` = 1 % ``modulus``
+/// Hence ``bezout_coefficient_a`` % ``modulus`` is the multiplicative inverse of ``num``
+///
+/// Before returning number is converted into a positive number
+///
+/// # Panics
+///
+/// This may return None when converting from f32 or f64,
+/// and will always succeed when converting from any integer or unsigned primitive, or ``BigUint``.
+#[must_use]
+pub fn mod_inverse(num: &BigUint, modulus: &BigUint) -> Option<BigInt> {
+    /// This computes the greatest common divisor (``gcd``) of integers ``num1`` and ``num2``,
+    /// and also the coefficients of Bézout's identity, which are integers ``bezout_coefficient_a`` and ``bezout_coefficient_b`` such that
+    /// ``num1`` * ``bezout_coefficient_a`` + ``num2`` * ``bezout_coefficient_b`` = ``gcd``(num1,num2)
+    #[inline]
+    fn extended_binary_gcd(num1: &BigUint, num2: &BigUint) -> (BigInt, BigUint) {
+        // Assume non zero
+        assert!(!num1.is_zero());
+        assert!(!num2.is_zero());
```

Comment:
I would recommend that we add an assert that num1 is strictly less than modulus. That will help us simplify the logic below.